src/backend/distributed/transaction/worker_transaction.c

```diff
@@ -639,3 +640,53 @@ ErrorIfAnyMetadataNodeOutOfSync(List *metadataNodeList)
 		}
 	}
 }
+
+
+/*
+ * ErrorIfWorkerDistNodeIsLocked checks if the current process has taken
+ * an ExlusiveLock on the pg_dist_node table of the worker and raises an error
+ * if that is the case
+ */
+void
+ErrorIfWorkerDistNodeIsLocked(WorkerNode *workerNode)
+{
+	int connectionFlags = REQUIRE_METADATA_CONNECTION;
+
+	MultiConnection *workerConnection = GetNodeUserDatabaseConnection(connectionFlags,
+																	  workerNode->
+																	  workerName,
+																	  workerNode->
+																	  workerPort,
+																	  CurrentUserName(),
+																	  NULL);
+	const char *partialCommand =
+		"SELECT COUNT(*) FROM pg_locks JOIN pg_class ON pg_locks.relation = pg_class.oid "
+		"WHERE pg_class.relname = 'pg_dist_node' AND pg_locks.granted AND pg_locks.mode = 'ExclusiveLock' AND pid = ";
```

Comment:
this seems a bit sketchy because this view is not always reliable

each node has a unique identifier in `pg_dist_node_metadata`. Can we check whether the identifier on the node we're adding is the same as our own?